clang/lib/AST/Expr.cpp

```diff
@@ -1332,6 +1334,11 @@ StringLiteral::getLocationOfByte(unsigned ByteNo, const SourceManager &SM,
                                  const LangOptions &Features,
                                  const TargetInfo &Target, unsigned *StartToken,
                                  unsigned *StartTokenByteOffset) const {
+  // No source location of bytes for binary literals since they don't come from
+  // source.
+  if (getKind() == StringLiteralKind::Binary)
+    return getStrTokenLoc(0);
```

Comment:
Does this mean that diagnostics involving invalid use of binary string literals will have no source location to point to?

Reply:
It will point at the beginning of the StringLiteral, but not to a particular problematic byte (which this function intends to find). In case of `#embed` it will point to the directive location. The test I'm adding should exercise that.